### PR TITLE
add drop_nonsampled; add ability to name sampling variable

### LINKED ISSUE #427: declare_sampling(filter = FALSE) option
Suggesting an option for declare_sampling that turns on/off the filter/subset step. When setting up sampling-dependent estimands that are *out of sample* you have to not use declare_sampling to get at them. This would be easy change I think.

### PR DESCRIPTION
**Pull request by graemeblair**
closes https://github.com/DeclareDesign/DeclareDesign/issues/427

built on @nfultz's two PR's #429 #428 but using a logical for dropping

### CODE REVIEW

**Comment by graemeblair** [comment]
will wait to merge pending @nfultz review

File: R/declare_sampling.R [hunk 1]
```diff
@@ -38,32 +38,35 @@
...
     ID_label = NA
   )
-
-  S <- as.character(S)
+  
+  S <- as.character(as.symbol(samp))
 
   ## subset to the sampled observations
-  data[ data[[S]] %in% 1, names(data) != S, drop = FALSE]
+  if(drop_nonsampled == TRUE) {
```
Comment (nfultz) [inline R/declare_sampling.R:65]: no `==TRUE`

**Review by nfultz — APPROVED** [review]
lgtm

**Comment by nfultz** [comment]
If you aren't using the drat repo anymore, you can probably remove the ddtools hook to fix the build.

**Comment by graemeblair** [comment]
Thanks @nfultz. We are still intermittently using it because some people can't upgrade their version of R.

**Pull request merged**